src/legacy/ui/public/notify/banners/banners.tsx

```diff
@@ -17,89 +17,43 @@
  * under the License.
  */
 
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { npStart } from 'ui/new_platform';
+import { I18nProvider } from '@kbn/i18n/react';
+
+const npBanners = npStart.core.overlays.banners;
+
+/** compatibility layer for new platform */
+const mountForComponent = (component: React.ReactElement) => (element: HTMLElement) => {
+  ReactDOM.render(<I18nProvider>{component}</I18nProvider>, element);
+  return () => ReactDOM.unmountComponentAtNode(element);
+};
+
 /**
  * Banners represents a prioritized list of displayed components.
  */
 export class Banners {
-
-  constructor() {
-    // sorted in descending order (100, 99, 98...) so that higher priorities are in front
-    this.list = [];
-    this.uniqueId = 0;
-    this.onChangeCallback = null;
-  }
-
-  _changed = () => {
-    if (this.onChangeCallback) {
-      this.onChangeCallback();
-    }
-  }
-
-  _remove = id => {
-    const index = this.list.findIndex(details => details.id === id);
-
-    if (index !== -1) {
-      this.list.splice(index, 1);
-
-      return true;
-    }
-
-    return false;
-  }
-
-  /**
-   * Set the {@code callback} to invoke whenever changes are made to the banner list.
-   *
-   * Use {@code null} or {@code undefined} to unset it.
-   *
-   * @param {Function} callback The callback to use.
-   */
-  onChange = callback => {
-    this.onChangeCallback = callback;
-  }
-
   /**
    * Add a new banner.
    *
    * @param {Object} component The React component to display.
    * @param {Number} priority The optional priority order to display this banner. Higher priority values are shown first.
-   * @return {String} A newly generated ID. This value can be used to remove/replace the banner.
+   * @return {symbol} A newly generated ID. This value can be used to remove/replace the banner.
```

Comment:
wouldn't it affect downstream code?

Reply:
It's used as an opaque identifier that isn't modified. I audited our usages and confirmed this won't be a problem. I'd rather use a symbol here to emphasize that this is opaque.

Reply:
After considering that we probably should use a unique string for the `key` prop to avoid any leftover data added to the `HTMLElement` the banner render into, I'm going to put this back to strings (since symbols can't be used as `key` props).